### PR TITLE
chore(sonar): fix code-scanner findings #437

### PR DESCRIPTION
**Pull request by ntruchsess**
## Description

* remove GET /api/user/app/{appId}/roles
* remove PUT /api/user/app/{appId}/roles
* findings from sonar were fixed
* findings from codeql were fixed

corresponding PR in portal CD-repository: https://github.com/eclipse-tractusx/portal/pull/210

## Why

removal of obsolete code and fixing code-scanner issues increases maintainability.

## Issue

#437 

## Checklist

Please delete options that are not relevant.

- [X] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [X] I have performed a self-review of my own code
- [X] I have successfully tested my changes locally
- [X] I have added tests that prove my changes work
- [X ] I have checked that new and existing tests pass locally with my changes

### CODE REVIEW

File: src/framework/Framework.Models/Encryption/CryptoHelper.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ ********************************************************************************/
+
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Org.Eclipse.TractusX.Portal.Backend.Framework.Models.Encryption;
+
+public static class CryptoHelper
+{
+    public static (byte[] Result, byte[] InitializationVector) Encrypt(string data, byte[] encryptionKey, CipherMode cipherMode, PaddingMode paddingMode)
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/framework/Framework.Models/Encryption/CryptoHelper.cs:27]: done within UnitTest: `EncryptDecrypt_WithIV_Success`

[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/framework/Framework.Models/Encryption/CryptoHelper.cs:27]: it's tested, see: https://github.com/eclipse-tractusx/portal-backend/blob/122884864aa2ce07d44868cfe4fbfd664e6aa4d2/tests/framework/Framework.Models.Tests/EncryptionHelperTests.cs#L59

File: src/portalbackend/PortalBackend.Migrations/Migrations/20240301164139_issue437-codeql-encryption.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.Migrations.Migrations
+{
+    /// <inheritdoc />
+    public partial class issue437codeqlencryption : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<int>(
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20240301164139_issue437-codeql-encryption.cs:32]: that should be done in a separate ticket. It needs to decrypt and re-encrypt all those entries. But as the mode being used is stored along with the entry such data-migration is independent from deployment of this PR

[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20240301164139_issue437-codeql-encryption.cs:32]: clarified

**Review by Phil91 — COMMENTED** [review]
Will approve as soon as CD-PR is there.

**Review by Phil91 — APPROVED** [review]

**Review by Phil91 — APPROVED** [review]

**Pull request merged**